test/Microsoft.AspNet.Mvc.ModelBinding.Test/Metadata/AssociatedMetadataProviderTest.cs

```diff
@@ -233,6 +233,42 @@ public void GetMetadataForProperties_SetsContainerAsExpected()
             Assert.Same(model, metadata[2].Container);
         }
 
+        [Fact]
+        public void GetMetadataForPropertiesFromModelMetadataType()
+        {
+            // Arrange
+            var model = new PropertyModel { LocalAttributes = 8, MetadataAttributes = "hello", MixedAttributes = 21.12 };
+            var provider = new TestableAssociatedMetadataProvider();
+
+            // Act
+            // Call ToList() to force the lazy evaluation to evaluate
+            provider.GetMetadataForProperties(model, typeof(PropertyViewModel)).ToList();
```

Comment:
also strange to pass unrelated types to `GetMetadataForProperties()`.  the model isn't assignable to the passed `Type`.